src/Servers/Kestrel/samples/PlaintextApp/Startup.cs

```diff
@@ -42,4 +45,22 @@ public static Task Main(string[] args)
             return host.RunAsync();
         }
     }
+
+    internal static class ValueTaskExtensions
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Task GetAsTask(this in ValueTask<FlushResult> valueTask)
```

Comment:
Seems like this could be submitted as an improvement to AsTask?

Reply:
Its converting a result type `ValueTask<T>` task-like to a non-result type `Task`

`AsTask()` would covert it to `Task<T>` would need a different name to work with overloading?

Reply:
Right, this would need to go into pipelines itself.